xonsh/built_ins.py

```diff
@@ -427,7 +427,15 @@ def run_subproc(cmds, captured=True):
         write_target = cmds[-1][0]
         write_mode = WRITER_MODES[cmds[-2]]
         cmds = cmds[:-2]
-        last_stdout = PIPE
+        if write_target is not None:
+            try:
+                last_stdout = open(write_target, write_mode)
```

Comment:
Ah; I had thought that subprocess would close this itself when it was done (since `close_fds` defaults to `True`), but I must have been misunderstanding.  I added a check to the end of `run_subproc` to close the file.